app/Services/BigCommerceService.php

```diff
@@ -0,0 +1,152 @@
+        }
+    }
+
+    public function getStoreSettings(): Collection
+    {
+        $xmlResponse = $this->makeBigCommerceAPIRequest(
+            'GET',
+            'stores/' . $this->getStoreHash() . '/v2/store'
+        );
+        return $this->createCollectionFromXml($xmlResponse);
+    }
+
+    public function deleteExistingScript() {
+        $scriptRow = Script::where('store_hash', $this->getStoreHash())->first();
+        if ($scriptRow) {
+            $uuid = $scriptRow['uuid'];
+            $this->makeBigCommerceAPIRequest(
+                'DELETE',
+                'stores/' . $this->getStoreHash() . '/v3/content/scripts/' . $uuid
+            );
+            $scriptRow->delete();
+        }
+    }
+
+    public function createScript(Config $config)
+    {
+        $jsSnippet = view('jsSnippet', [
+            'shopkey' => $config->shopkey
+        ]);
+
+        $requestBody = [
+            'name' => 'Findologic',
+            'description' => 'Search & Navigation Platform',
+            'html' => $jsSnippet->render(),
+            'auto_uninstall' => true,
+            'load_method' => 'default',
+            'location' => 'head',
+            'visibility' => 'storefront',
+            'kind' => 'script_tag',
+            'consent_category' => 'essential',
+        ];
+
+        $response = $this->makeBigCommerceAPIRequest(
+            'POST',
+            'stores/' . $this->getStoreHash() . '/v3/content/scripts',
+            json_encode($requestBody)
+        );
+
+        $collection = collect(json_decode($response->getBody(), true));
+        $data = $collection['data'];
+
+        $script = new Script();
+        $script->store_id = $config->store_id;
+        $script->name = $data['name'];
+        $script->uuid = $data['uuid'];
+        $script->store_hash = $this->getStoreHash();
+        $script->save();
+    }
+
+    private function makeBigCommerceAPIRequest($method, $endpoint, $body = ''): ResponseInterface
+    {
+        $requestConfig = [
+            'headers' => [
+                'X-Auth-Client' => env('CLIENT_ID'),
+                'X-Auth-Token' => $this->getAccessToken(),
+                'Content-Type' => 'application/json',
+            ]
+        ];
+
+        if (!empty($body)) {
+            $requestConfig['body'] = $body;
+        }
+
+        try {
+            return $this->httpClient->request($method, 'https://api.bigcommerce.com/' . $endpoint, $requestConfig);
+        } catch (RequestException $e) {
+            return $e->getResponse();
+        }
+    }
+
+    private function createCollectionFromXml($xmlResponse): Collection
+    {
+        if ($xmlResponse->getStatusCode() >= 200 && $xmlResponse->getStatusCode() < 300) {
+            $xml = simplexml_load_string($xmlResponse->getBody(), 'SimpleXMLElement', LIBXML_NOCDATA);
+
+            $json = json_encode($xml);
+            $array = json_decode($json, true);
+            $collection = collect($array);
```

Comment:
This does not need to be stored inside a variable, you can simply do `return collect($array)`